app/controllers/course/assessment/question/programming_controller.rb

```diff
@@ -105,7 +119,24 @@ def update_question_setting
   end
 
   def destroy
-    if @programming_question.destroy
+    koditsu_question_id = @programming_question.acting_as.koditsu_question_id
+
+    success = @programming_question.class.transaction do
+      raise ActiveRecord::Rollback unless @programming_question.destroy
+
+      is_course_koditsu_enabled = current_course.component_enabled?(Course::KoditsuPlatformComponent)
+
+      if is_course_koditsu_enabled && @assessment.is_koditsu_enabled
+        @assessment.update!(is_synced_with_koditsu: false)
+        remove_question_from_assessment_in_koditsu(koditsu_question_id)
```

Comment:
> > Any issue if Koditsu is not responding? i.e. no status code returned, instead, there was a timeout.
> 
> Once I got the issue (although quite rarely), it will raise an error and the operation is marked as failed (will do refactoring regarding this after PE, since need to think carefully on the spec should this happen)

There is nothing to think about on the spec: Coursemology should not fail if Koditsu service fails.

Reply:
done.. having the rescue mechanism already for the case of Koditsu service failing, due to one reason or another